default/AI/ResourcesAI.py

```diff
@@ -211,30 +376,225 @@ def assess_protection_focus(pid):
+                    print ("Tried setting %s for Concentration Camp planet %s (%d) with species %s and current focus %s, got result %d and focus %s" %
+                           (pinfo.future_focus, planet.name, pid, planet.speciesName, pinfo.current_focus, True, planet.focus))
+                print "%s focus of planet %s (%d) (with Concentration Camps/Remnants) at Industry Focus" % (["set", "left"][pinfo.current_focus == IFocus], planet.name, pid)
+                continue
+            else:
+                newplanet = universe.getPlanet(pid)
+                print ("Error: Failed setting %s for Concentration Camp planet %s (%d) with species %s and current focus %s, but new planet copy shows %s" %
+                       (pinfo.future_focus, planet.name, pid, planet.speciesName, planet.focus, newplanet.focus))
+
+
+def set_planet_protection_foci(focus_manager):
+    """Assess and set protection foci"""
+    universe = fo.getUniverse()
+    for pid, pinfo in focus_manager.raw_planet_info.items():
+        planet = pinfo.planet
+        if PFocus in planet.availableFoci and assess_protection_focus(pid, pinfo):
+            curFocus = planet.focus
+            if focus_manager.bake_future_focus(pid, PFocus):
+                if curFocus != PFocus:
+                    print ("Tried setting %s for planet %s (%d) with species %s and current focus %s, got result %d and focus %s" %
+                           (pinfo.future_focus, planet.name, pid, planet.speciesName, curFocus, True, planet.focus))
+                print "%s focus of planet %s (%d) at Protection(Defense) Focus" % (["set", "left"][curFocus == PFocus], planet.name, pid)
+                continue
+            else:
+                newplanet = universe.getPlanet(pid)
+                print ("Error: Failed setting %s for planet %s (%d) with species %s and current focus %s, but new planet copy shows %s" %
+                       (focus_manager.new_foci[pid], planet.name, pid, planet.speciesName, planet.focus, newplanet.focus))
+
+
+def set_planet_happiness_foci(focus_manager):
+    """Assess and set planet focus to preferred focus depending on happiness"""
+    # TODO Assess need to set planet to preferred focus to improve happiness
+
+
+def set_planet_industry_and_research_foci(focus_manager, priorityRatio):
+    """Adjust planet's industry versus research focus while targetting the given ratio and avoiding penalties from changing focus"""
+    print "\n-----------------------------------------"
+    print "Making Planet Focus Change Determinations\n"
+
+    ratios = []
+    # for each planet, calculate RP:PP value ratio at which industry/Mining focus and research focus would have the same total value, & sort by that
+    # include a bias to slightly discourage changing foci
+    curTargetPP = 0.001
+    curTargetRP = 0.001
+    resource_timer.start("Loop")  # loop
+    has_force = tech_is_complete("CON_FRC_ENRG_STRC")
+    # cumulative all industry focus
+    ctPP0, ctRP0 = 0, 0
```

Comment:
I plan to cleanup variable names and formatting myself.  It is quite easy with IDE refactor tools and have less chances to got mistake.